makes the z9 rifle more usable + sprite edits

gy1ta23 opened the pull request:
🆑
rscadd: the Z9 rifle has its own mags, which can now be ordered, and printed at a hacked lathe
/🆑

I love the Z9. I think it's great. But it is, as of right now, nearly unusable as an ordered weapon. Chiefly, you can't order magazines for it, as it doesn't have magazines; you need to print assault rifle magazines. This is crippling. I decided to stray from the Z8 and order Z9s to deal with raiders one XO round and I had to order four rifle crates and cannibalize half of the rifles so the other half would have one more magazine to shoot. It's a very good thing we didn't have to actually use them.

As such, I've added a unique magazine for it, plus crates to buy them and practice munitions for practice. I've done some hackjob sprite editing to give it this magazine; if you think it's too long, that can be fixed, although I think considering the magazine capacity and cartridge it makes sense, as it's about the same as the assault rifle magazine it previously fed from minus the curve. I also noticed that the Z9's onmobs were very strange, and the size (or really, length) of it wasn't consistent across the onmobs or the item sprite, so I shortened it so it has more of a visually unique profile compared to the Z8 and is more congruent with the southern/northern facing onmob sprites.

Here's an [imgur album](https://imgur.com/a/X5X8gnS) with all the new stuff.

I also fixed some typos in both bullpup rifle descriptions, because I'd have conniptions if I didn't. Also, also, bullpup magazines have been named "bullup" magazines for four years now in the sprite files and I took this as my chance to fix it.

juxjux9930 commented on code/modules/projectiles/ammunition/boxes.dm:
```diff
@@ -332,6 +332,22 @@
 	ammo_type = /obj/item/ammo_casing/rifle/military/practice
 
+/obj/item/ammo_magazine/mil_rifle/light
+	name = "assault rifle magazine"
```
```suggestion
```

juxjux9930 commented on code/modules/projectiles/ammunition/boxes.dm:
```diff
@@ -332,6 +332,22 @@
+	matter = list(MATERIAL_STEEL = 1800)
+	ammo_type = /obj/item/ammo_casing/rifle
+	max_ammo = 20
+	multiple_sprites = 1
```
```suggestion
```

juxjux9930 commented on code/modules/projectiles/ammunition/boxes.dm:
```diff
@@ -332,6 +332,22 @@
+	name = "assault rifle magazine"
+	icon_state = "bullpup_light"
+	caliber = CALIBER_RIFLE
+	matter = list(MATERIAL_STEEL = 1800)
```
```suggestion
```

juxjux9930 reviewed (changes requested):
The light bullpup magazine would inherit these three from the regular magazine

MuckerMayhem commented:
Not sure how to feel about adding another entry to a pool of already hard-to-understand and convoluted ammo types. I personally feel like a better way of solving this issue is to just add ammo crates that can be ordered, or just printing off magazines you need in-game and ordering the base materials you need to do that for lathes.

gy1ta23 committed:
rifles!!!1


fixes descs


lathemags


oops i forgot a mag


holy shit hitting / is not that hard


Update code/modules/projectiles/ammunition/boxes.dm

Co-authored-by: Jux <68120725+juxjux9930@users.noreply.github.com>
Update code/modules/projectiles/ammunition/boxes.dm

Co-authored-by: Jux <68120725+juxjux9930@users.noreply.github.com>
Update code/modules/projectiles/ammunition/boxes.dm

Co-authored-by: Jux <68120725+juxjux9930@users.noreply.github.com>

MuckerMayhem reviewed (approved):
Works for me.
